### ISSUE TITLE
Add confirmation on Goal update

### ISSUE THREAD

**Issue by tijlleenders** [tijlleenders/ZinZen#1916]
**Additional context**
This was previously implemented on `main` branch thanks to @Shubham-Praj , just has to be ported to `develop`.

**Is your feature request related to a problem? Please describe.**
It's very annoying when you don't get confirmation of your changes. After Goal update the edit modal just closes.

**Describe the solution you'd like**
1. A very brief Toast message 'Goal updated!'
2. A very short (compared to add Goal) sound
Not implemented in `main` - but should be implemented now: if there are no changes, don't show the modal.

**Describe alternatives you've considered**
A confirmation popup. This breaks the flow.

**Additional context**
None.

**Comment by Shubham-Praj** [comment]
Hii @tijlleenders please assign this to me. Thanks

**Comment by tijlleenders** [comment]
On a roll @Shubham-Praj  :)

**Comment by tijlleenders** [comment]
Still requires a PR to the 'develop' branch, where a major refactor is taking place.

**Comment by Nicholas-Kipkoech** [comment]
@tijlleenders Was this done?

**Comment by tijlleenders** [comment]
@Nicholas-Kipkoech You can double check.

**Comment by tijlleenders** [comment]
@vinaybadgujar102 Did you see there was already a PR?

**Comment by vinaybadgujar102** [comment]
> @vinaybadgujar102 Did you see there was already a PR?

@tijlleenders , I noticed that the issue was added to the backlog, so I picked it up and started working on it. However, I missed that a PR had already been created for it. Sorry for missing that the PR was already there.

As you mentioned, the PR is already made, so I checked it and then reviewed the approach. The existing PR takes a brute-force approach, comparing every field between the original and updated versions. In contrast, the approach I used involves creating a hash for both the original and the updated goal objects and then comparing the hashes to detect any differences.
